x-pack/plugins/security_solution/public/detections/components/alerts_kpis/common/hooks.ts

```diff
@@ -95,14 +97,22 @@ export function getAggregatableFields(
 
 export const useStackByFields = (useLensCompatibleFields?: boolean) => {
   const { pathname } = useLocation();
-
+  const { addError } = useAppToasts();
   const { browserFields } = useSourcererDataView(getScopeFromPath(pathname));
-  const allFields = useMemo(() => getAllFieldsByName(browserFields), [browserFields]);
-  const [stackByFieldOptions, setStackByFieldOptions] = useState(() =>
-    getAggregatableFields(allFields, useLensCompatibleFields)
-  );
-  useEffect(() => {
-    setStackByFieldOptions(getAggregatableFields(allFields, useLensCompatibleFields));
-  }, [allFields, useLensCompatibleFields]);
-  return useMemo(() => stackByFieldOptions, [stackByFieldOptions]);
+
+  return useCallback(() => {
+    try {
+      return getAggregatableFields(getAllFieldsByName(browserFields), useLensCompatibleFields);
+    } catch (err) {
+      addError(err, {
```

Comment:
@paulewing what do you think of this message? Looks like:
<img width="352" alt="Screenshot 2023-10-09 at 3 02 18 PM" src="https://github.com/elastic/kibana/assets/6935300/62ac44fd-d9c8-4ab7-990b-9b2a8397252c">